### PR TITLE
fix(echo): Replace faulty instanceof check with static code check

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?
* The `error instanceof EchoError` check was evaluating to `false` even when passed a valid `EchoError`. The issue only appears with the CommonJS bundle, not with ESModules. The decision was made to avoid `instanceof` altogether and replace with a static check to identify if the `Error.code` matches an expected code
* This fixes [this failing test](https://github.com/novuhq/novu/actions/runs/9367163429/job/25786344943?pr=5675#step:10:448) 

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/create-novu-app/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   "scripts": {
     "start": "node dist/index.js",
     "dev": "ncc build ./index.ts -w -o dist/",
-    "prerelease": "node ../../scripts/rm.mjs dist",
+    "prerelease": "rimraf dist",
```
Comment (rifont) [inline packages/create-novu-app/package.json:23]: Small fix to ensure installation passes without error.

File: packages/echo/src/handler.ts [hunk 2]
```diff
@@ -261,8 +262,12 @@ export class EchoRequestHandler<Input extends any[] = any[], Output = any> {
     }
   }
 
-  private handleError(error: any): IActionResponse {
-    if (error instanceof EchoError) {
```
Comment (djabarovgeorge) [inline packages/echo/src/handler.ts:265]: 🔥 the instanceof in cross bundles

**Review by djabarovgeorge — APPROVED** [review]
🚀

File: docker/local/docker-compose.e2e.yml
Comment (rifont) [inline docker/local/docker-compose.e2e.yml:1]: This addition was needed to fix the failing CI pipeline, it was accidentally removed in a previous PR.

**Pull request merged**